pkg/operator/controller/externaldns/pod.go

```diff
@@ -325,30 +328,33 @@ func (b *externalDNSContainerBuilder) fillAWSFields(container *corev1.Container)
 
 	env := []corev1.EnvVar{
 		{
-			Name: awsAccessKeyIDEnvVar,
+			Name: awsSharedCredentialsFile,
 			ValueFrom: &corev1.EnvVarSource{
 				SecretKeyRef: &corev1.SecretKeySelector{
 					LocalObjectReference: corev1.LocalObjectReference{
 						Name: b.secretName,
 					},
-					Key: awsAccessKeyIDKey,
-				},
-			},
-		},
-		{
-			Name: awsAccessKeySecretEnvVar,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: b.secretName,
-					},
-					Key: awsAccessKeySecretKey,
+					Key: awsSharedCredentialsFile,
 				},
 			},
 		},
 	}
 
 	container.Env = append(container.Env, env...)
+
+	for _, v := range b.volumes {
+		if v.Name == awsCredentialsVolumeName {
+			container.Env = append(container.Env, corev1.EnvVar{
```

Comment:
The environment variable is being appended here again. This one looks correct compared to the previous one.

Reply:
Removed this one.